tom/tag.py

```diff
@@ -63,6 +63,9 @@ def get_next_build_tag(self, old_tag, branch):
                 raise TagParsingException('Could not split tag [%s] into parts' % old_tag)
         else:
             # 3.15.0 -> 3.16.0-build1 or 3.15.1-build1, depending on branch
+            # But first, strip release number from version
+            # 3.12.5-2 -> 3.12.5
+            old_tag = re.sub('-[0-9]+','',old_tag)
```

Comment:
Would `-[0-9]+$` work to make sure we only strip the end of the string?

Reply:
well, the thing is that code that follows (`get_next_version`) expects a string of three dot-separated numbers. So proper way is to strip only the proper stuff (`-[0-9]+$`), and fail with a nice message when met with something unexpected.